### ISSUE TITLE
ReadableStream.prototype.pipeTo's first argument must be a WritableStream

### ISSUE THREAD

**Issue by ibudisteanu** [transcend-io/conflux#64]
Trying to follow example 1 https://github.com/transcend-io/conflux/blob/master/example/pipes/index.js in browser and I get `ReadableStream.prototype.pipeTo's first argument must be a WritableStream`. Any idea ?

**Comment by jimmywarting** [comment]
It might be possible that you are mixing both native and polyfilled whatwg streams?

eg: you are getting a native readable stream from `Response.prototype.body` and tries to pipe the response to a polyfilled `WritableStream` from web-streams-polyfill?

(they are not compatible with eachother)

**Comment by ibudisteanu** [comment]
Hi Jimmy! I love the StreamSaver and I have already integrated it in my research code, but I can't integrate the zip. 

I need to use your library because we believe that most of streams will be big ( > 4gb maybe 10 gb). These binary data (files) in the browser will be stored locally (index db) into chunks (ex: 32 mb chunk). I was able to use the StreamSaver to saveAs one of these files in the browser, but I simply can't integrate the conflux (we need it because we have also directories as well).

Here is a snippet of my entire code.

```

        const streamer = streamsaver.createWriteStream( name + '.zip', )

        const {readable, writable } = new Writer();
        const writer = writable.getWriter();

        let stopped = false;

        async.eachLimit( all_streams_to_save, 1, ( current_stream_to_save, next) => {
       
                const mystream = new stream.PassThrough();

                async.each( current_stream_to_save.chunks, ( chunkIndex, next2 )=>{

                    getChunk(current_stream_to_save, chunkIndex, (err, buffer) => {

                        mystream.write(buffer);
                        next2();

                    });

                }, (err, out ) => {

     
                     // Add a file
                     writer.write({
                            name: current_stream_to_save.path,
                            lastModified: new Date(0),
                            stream: mystream,
                    });

                    mystream.end();

                    next();

                });

          

        }, (err) => {

            readable.pipeTo(streamer);
            writer.close();
            cb(null, { done: true } )

        } );

```

**Comment by ibudisteanu** [comment]
I believe the error is here

`const mystream = new stream.PassThrough();`

How can create a stream compatible  with conflux created dynammically from buffers?

**Comment by ibudisteanu** [comment]
```
readable.pipeTo(streamer);
```
throws the following error

![image](https://user-images.githubusercontent.com/17750489/89078421-11297300-d38d-11ea-9251-01b8762bd7a4.png)

and nothing happens.

**Comment by ibudisteanu** [comment]
@jimmywarting  helped me out with great communications and support! He was able to identity the issue and refactor my code to make it working. 

First  issue was that we used native node buffers instead of using whatwg streams.

The 2nd issue was that we had to use web-streams-polyfill to make it work.
https://github.com/Pandora-Protocol/pandora-protocol-reference/issues/6#issuecomment-667509501

Here is the working code https://github.com/Pandora-Protocol/pandora-protocol-reference/blob/7f13e460d64060e7b73a06cdb40ce65a2c7b69ac/src/pandora-locations/browser/browser-pandora-locations.js#L111

Thank you so much again for taking the time to support the research code we do and for integrating conflux into our code.
